app/src/main/java/com/example/healthinspector/Fragments/CartFragment.java

```diff
@@ -30,6 +44,30 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
     @Override
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
+        ArrayList<RecommendedProduct> cartItems = new ArrayList<>();
+        ParseQuery<Cart> parseQuery = ParseQuery.getQuery(Cart.class);
+        try {
+            Cart userCart = parseQuery.get(ParseUser.getCurrentUser().getParseObject(Constants.CART).getObjectId());
+            for(int i = 0; i < userCart.getCartItems().length(); i++){
+                //create new recommendedProduct object and add to arraylist
+                String keyWords = userCart.getCartItems().getJSONObject(i).getString(Constants.KEYWORDS);
+                String productImageUrl = userCart.getCartItems().getJSONObject(i).getString(Constants.PRODUCT_IMAGE_URL);
+                String brand = userCart.getCartItems().getJSONObject(i).getString(Constants.BRAND);
+                String productName = userCart.getCartItems().getJSONObject(i).getString(Constants.PRODUCT_NAME);
+                JSONArray nutrientLevelsJSON = userCart.getCartItems().getJSONObject(i).getJSONArray(Constants.NUTRIENT_LEVELS);
```

Comment:
nit: store `userCart.getCartItems().getJSONObject(i)` in a variable and then use it